features/step_definitions/shopping_cart_steps.rb

```diff
@@ -0,0 +1,27 @@
+When("I click on {string} on {string}") do |link, title|
+  @product = Product.find_by(title: title)
+    within("#product-#{@product.id}") do
+      click_link_or_button link
+    end
+end
+
+Then("a cart should be created for me") do
+  @cart = Cart.last
+  expect(@cart).to_not be nil
+end
+
+Then("the cart should contain {string}") do |string|
+  expect(@cart.shopping_cart_items.first.item).to eq @product
+end
+
+Given("there is a cart with cart item {string}") do |product|
+  steps %Q{
+       And I click on "Add to cart" on "#{product}"
+       Then a cart should be created for me
+       And the cart should contain "#{product}"
```

Comment:
I don't see a need to include these 2 assertions in this step definition.

Reply:
Ok, I will remove them

Reply:
Removed
```And the cart should contain "#{product}"```
But
```Then a cart should be created for me```
is necessary for the acceptance test `Given there is a cart with cart item "Mug"` to pass

Reply:
Tested the ternary button, no need for 
`Then a cart should be created for me`
anymore. I will remove it.